python/cudf/cudf/core/column/numerical.py

```diff
@@ -412,6 +413,14 @@ def find_and_replace(
         """
         Return col with *to_replace* replaced with *value*.
         """
+        to_replace_col = as_column(to_replace)
+        replacement_col = as_column(replacement)
+
+        if not isinstance(to_replace_col, NumericalColumn) or not isinstance(
+            replacement_col, NumericalColumn
+        ):
+            return self
```

Comment:
Yes, we would have to return a deep copy here. But if I do so, mypy seems to be giving me this error:

```
python/cudf/cudf/core/column/numerical.py:422: error: Incompatible return value type (got "ColumnBase", expected "NumericalColumn")
Found 1 error in 1 file (checked 186 source files)
```

Do we need to create a Union of NumericalColumn and ColumnBase as the return type in this case? 

cc: @shwina

Reply:
I'm guessing this is because `.copy` is inherited from `ColumnBase`?

Reply:
Yes -- right now [`ColumnBase.copy()` returns a `ColumnBase`](https://github.com/rapidsai/cudf/blob/branch-0.18/python/cudf/cudf/core/column/column.py#L517), but I think it should instead return the same type as `self`, similar to [`take`](https://github.com/rapidsai/cudf/blob/branch-0.18/python/cudf/cudf/core/column/column.py#L805).

Reply:
Got it, after updating the typings, looks like our `build_column` has return typing as `ColumnBase`, which is why I'm getting this error:

```
python/cudf/cudf/core/column/column.py:525: error: Incompatible return value type (got "ColumnBase", expected "T")
```

But this shouldn't even come up no, because T is a TypeVar with bound as "ColumnBase" specified in column.py file?

Reply:
Similar to `take`, you have to use `cast` to cast the result of `build_column` to a `T`. This is far from ideal, but we have to figure out a way to write overloads for `build_column` based on its dtype argument. Until then, casting is fine. Note that `cast` should have no effect on runtime behaviour or performance

Reply:
Got it, thanks 👍